zerocopy-derive/src/enum.rs

```diff
@@ -147,6 +153,14 @@ fn generate_variant_structs(
             ) #where_clause;
         };
 
+        let invariants_eq_impl: TokenStream = parse_quote! {
+            // SAFETY: TODO
+            //
+            // TODO: This might not actually be sound – it's possible for there
+            // to be `UnsafeCell`s in some but not all enum variants.
+            unsafe impl #impl_generics #zerocopy_crate::pointer::InvariantsEq<#variant_struct_ident #ty_generics> for ___ZerocopyVariants #ty_generics #where_clause {}
```

Comment:
We might want to just replace `BecauseInvariantsEq` with `BecauseImmutable` and then impl `Immutable` for `___ZerocopyVariants` and  each`___ZerocopyVariantStruct_Xxx`. We currently require all unions to be `Immutable`, so this should be fine – we can just add an `OuterEnum: Immutable` bound to each `Immutable` impl block (where `OuterEnum` is the name of the type being derived on).